ui/appui-react/src/appui-react/framework/FrameworkDialogs.ts

```diff
@@ -0,0 +1,101 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { DialogChangedEvent, DialogInfo, DialogManagerBase } from "../dialog/DialogManagerBase";
+
+/** Modal Dialog Changed Event class.
+ * @public
+ */
+export class ModalDialogChangedEvent extends DialogChangedEvent { }
+
+/** Modeless Dialog Changed Event class.
+ * @public
+ */
+export class ModelessDialogChangedEvent extends DialogChangedEvent { }
+
+/** @public */
+export interface ModelessDialogInfo {
+  reactNode: React.ReactNode;
+  zIndex: number;
+  parentDocument: Document;
+}
+
+/**
+ * Manages dialog access
+ * @beta
+ */
+export interface FrameworkDialog {
+  /** @internal */
+  readonly dialogManager: DialogManagerBase;
+
+  /** Get the array of modal dialogs */
+  readonly dialogs: DialogInfo[];
+
+  /** Open a modal dialog
+   * @param dialog The Dialog to open
+   * @param id The id of the Dialog. If one is not provided, an id is generated.
+   * @param parentDocument The Document used to determine the owning window.
+   */
+  openDialog(dialog: React.ReactNode, id?: string, parentDocument?: Document): void;
+
+  /** Close a modal dialog
+   * @param dialog The Dialog to open. If one is not specified, the active dialog will be closed.
+   */
+  closeDialog(dialog?: React.ReactNode): void;
+
+  /** @internal */
+  closeAll(): void;
+
+  /** Update the dialogs */
+  update(): void;
```

Comment:
Do we plan to refactor these APIs later? I don't think this is needed in any way. Also feels like we should have one dialog manager w/ a `modeless` flag/options instead..